From 1578e778a27f442f472afd927bab9fb782cc4479 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Mon, 20 Feb 2017 07:00:33 +0000
Subject: [PATCH] phy: phy-bcm-ns2-usbdrd: Replace deprecated API of extcon

Based on upstream:
commit 830ae442202e (extcon: Remove the deprecated extcon functions)

This patch removes the deprecated notifier API of extcon framework and then use
the new extcon API[2] with the unique id[1] to indicate the each external
connector. Alter deprecated API as following:
    - extcon_register_interest() -> extcon_register_notifier()
    - extcon_unregister_interest() -> extcon_unregister_notifier()

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/phy/phy-bcm-ns2-usbdrd.c |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/phy/phy-bcm-ns2-usbdrd.c b/drivers/phy/phy-bcm-ns2-usbdrd.c
index ade7ad3..4c045ec 100644
--- a/drivers/phy/phy-bcm-ns2-usbdrd.c
+++ b/drivers/phy/phy-bcm-ns2-usbdrd.c
@@ -384,16 +384,14 @@ static int register_extcon_notifier(struct ns2_phy_driver *phy_driver,
 	edev = phy_driver->edev;
 
 	/* Register for device change notification */
-	ret = extcon_register_interest(&phy_driver->extcon_dev, edev->name,
-				       "USB", &phy_driver->dev_nb);
+	ret = extcon_register_notifier(edev, EXTCON_USB, &phy_driver->dev_nb);
 	if (ret < 0) {
 		dev_err(dev, "can't register extcon_dev for %s\n", edev->name);
 		return ret;
 	}
 
 	/* Register for host change notification */
-	ret = extcon_register_interest(&phy_driver->extcon_host, edev->name,
-				       "USB-HOST", &phy_driver->host_nb);
+	ret = extcon_register_notifier(edev, EXTCON_USB_HOST, &phy_driver->dev_nb);
 	if (ret < 0) {
 		dev_err(dev, "can't register extcon_dev for %s\n", edev->name);
 		goto err_dev;
@@ -420,9 +418,9 @@ static int register_extcon_notifier(struct ns2_phy_driver *phy_driver,
 	return 0;
 
 err_host:
-	extcon_unregister_interest(&phy_driver->extcon_dev);
+	extcon_unregister_notifier(edev, EXTCON_USB_HOST, &phy_driver->dev_nb);
 err_dev:
-	extcon_unregister_interest(&phy_driver->extcon_host);
+	extcon_unregister_notifier(edev, EXTCON_USB, &phy_driver->dev_nb);
 	return ret;
 }
 
-- 
1.7.5.4

